### PR TITLE
Add info about setuptools_scm

### PR DESCRIPTION
**Pull request by connortann**
Changes can be previewed here:

https://resqpy.readthedocs.io/en/docs/CONTRIBUTING.html#making-a-release

### CODE REVIEW

File: docs/CONTRIBUTING.rst [hunk 1]
```diff
@@ -262,6 +262,38 @@ Following [semantic versioning](https://semver.org/), increment the:
...
+
+When working with a development version of the code that does not correspond to
+a tagged release, the version number will look a little different, for example
+``1.6.2.dev301+gddfbf6c``.
+
+This can be interpreted as:
+
+* ``1.6.2`` : is the *next* expected release. The previous release would be ``1.6.1``.
+* ``dev301`` : 301 commits added since the previous release.
+* ``+gddfbf6c`` : a ``+g`` prefix followed by current commit ID: ``dfbf6c``.
```
Comment (maxm17) [inline docs/CONTRIBUTING.rst:284]: Should commit ID not be 'ddfbf6c'? Looks all good apart from that

**Review by maxm17 — CHANGES_REQUESTED** [review]
One minor change that I can see

File: docs/CONTRIBUTING.rst (hunk 1 above)
Comment (connortann) [reply docs/CONTRIBUTING.rst:284]: Thanks - fixed.

**Comment by connortann** [comment]
Added a slight refactor of the logic in resqpy.init .

**Pull request merged**